content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -154,9 +154,157 @@ TODO.
+
+```rust
+let audio_settings = asset_server.load("data/audio.json");
+//  ^^^^^^^^^^^^^^                ^^^^
+//  | This has the type           |
+//  | Handle<AudioSettings>       |
+//                                | Which is passed into load...
+
+impl AssetLoader for AudioSettingsAssetLoader {
+    type Asset = AudioSettings;
+//               ^^^^^^^^^^^^^
+//               | ...and checked against AssetLoader::Asset
+
+    /* snip */
+}
+```
+
+Now, when loading, first the `AssetServer` attempts to infer type of asset by the type of `Handle` to return. If that can't be done (for example, labelled paths are permitted to return different asset types), then as a fallback the file extension is checked just like before.
+
+```rust
+// This will be inferred from context to be a Gltf asset, ignoring the file extension
+let gltf_handle = asset_server.load("models/cube/cube.gltf");
+
+// This still relies on file extension due to the label
+let cube_handle = asset_server.load("models/cube/cube.gltf#Mesh0/Primitive0");
+```
+
+### File extensions are now optional
+
+Since the asset type can be used to infer the loader, neither the file to be loaded nor the `AssetLoader` need to have file extensions.
+
+```rust
+pub trait AssetLoader: Send + Sync + 'static {
+    /* snip */
+
+    /// Returns a list of extensions supported by this [`AssetLoader`], without the preceding dot.
+    fn extensions(&self) -> &[&str] {
+        // A default implementation is now provided
+        &[]
+    }
+}
+```
+
+Previously, an asset loader with no extensions was very cumbersome to use. Now, they can be used just as easily as any other loader. Likewise, if a file is missing its extension, Bevy can now choose the appropriate loader.
+
+```rust
+let license = asset_server.load::<Text>("LICENSE");
+```
+
+Appropriate file extensions are still recommended for good project management, but this is now a recommendation rather than a hard requirement.
+
+### Multiple `AssetLoader`'s can be selected for the same asset
+
+Now, a single path can be used by multiple asset handles as long as they are distinct asset types.
+
+```rust
+// Load the sound effect for playback
+let bang = asset_server.load::<AudioSource>("sound/bang.ogg");
+
+// Load the raw bytes of the same sound effect (e.g, to send over the network)
+let bang_blob = asset_server.load::<Blob>("sound/bang.ogg");
+
+// Returns the bang handle since it was already loaded
+let bang_again = asset_server.load::<AudioSource>("sound/bang.ogg");
+```
+
+Note that the above example uses turbofish syntax for clarity. In practice, it's not required, since the type of asset loaded can usually be inferred by surrounding context at the call site.
+
+```rust
+fn setup(mut effects: ResMut<SoundEffects>, asset_server: Res<AssetServer>) {
+    effects.bang = asset_server.load("sound/bang.ogg");
+    effects.bang_blob = asset_server.load("sound/bang.ogg");
+}
```

Comment:
```suggestion
#[derive(Resource)]
struct SoundEffects {
    bang: AudioSource,
    bang_blog: Blob,
}

fn setup(mut effects: ResMut<SoundEffects>, asset_server: Res<AssetServer>) {
    effects.bang = asset_server.load("sound/bang.ogg");
    effects.bang_blob = asset_server.load("sound/bang.ogg");
}
```

Would suggest the definition of `SoundEffect` be included in the snippet. If you're going to the effort of making this specific example it should be super obvious that these are different types, not just different fields/variable names :)

Reply:
Good call!